amd/src/courseformat/courseindex/courseindex.js

```diff
@@ -46,6 +68,55 @@ export default class Component extends BaseComponent {
         });
     }
 
+    /**
+     * Refresh the section list.
+     *
+     * @param {object} param
+     * @param {Object} param.element
+     */
+    _refreshCourseSectionlist({element}) {
+        const topsectionslist = element.firstsectionlist ?? [];
+        this._fixOrder(this.element, topsectionslist, this.topsections);
+
+        for (let p in element.secondsectionlist) {
+            let secondorder = element.secondsectionlist[p];
```

Comment:
I think this should probably be copied to avoid changing shared data.
```suggestion
            let secondorder = Array.from(element.secondsectionlist[p]);
```